hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndexUtils.java

```diff
@@ -395,7 +395,7 @@ public static <R> HoodieRecord<R> createNewTaggedHoodieRecord(HoodieRecord<R> ol
    * The actual partition name is contained in the index definition.
    */
   public static String getPartitionNameFromPartitionType(MetadataPartitionType partitionType, HoodieTableMetaClient metaClient, String indexName) {
-    if (MetadataPartitionType.FUNCTIONAL_INDEX.equals(partitionType)) {
+    if (MetadataPartitionType.FUNCTIONAL_INDEX.equals(partitionType) || MetadataPartitionType.SECONDARY_INDEX.equals(partitionType)) {
```

Comment:
Can this conversion logic be put into `MetadataPartitionType` for each type, by providing a new API
```
public enum MetadataPartitionType {
  ...
  FUNCTIONAL_INDEX(HoodieTableMetadataUtil.PARTITION_NAME_FUNCTIONAL_INDEX_PREFIX, "func-index-", -1) {
    ...
    @Override
    public String getPartitionPath(...) {
      return metaClient.getIndexMetadata().get().getIndexDefinitions().get(indexName).getIndexName();
    }
    ...
  },
  ...
  public String getPartitionPath(...) {
    ...
  }
}
```

Reply:
In this way, we can avoid the confusion of what partition name or path to use, by consolidating the logic into this new API.